manifests/repomd_signatures.pp

```diff
@@ -0,0 +1,21 @@
+# Generate or regenerate Yum repository metadata GPG signatures
+class pulp::repomd_signatures {
+  if !$::pulp::yum_regenerate_repomd_signatures {
+    $existing_sig_test = '[ -e repomd.xml.asc ] || '
+  }
+  $regen_command = "/usr/bin/sudo -u apache -H /usr/bin/bash -c '
+    cd /var/lib/pulp/published/yum/master/yum_distributor/ 2>/dev/null || exit 0
+    for d1 in `ls` ; do
+      pushd \$d1
+      for d2 in `ls` ; do
+        pushd \$d2/repomd >/dev/null
+        ${existing_sig_test}gpg --yes --detach-sign --armor repomd.xml
```

Comment:
I wonder about `gpg` vs `gpg2`.

Reply:
On RHEL7, Ubuntu 18.04, and other current systems, gpg is the same as gpg2, so it shouldn't matter.
On older systems, the command in puppet-pulp should match the command used in pulp_rpm, which is currently `gpg` not `gpg2`:
https://github.com/pulp/pulp_rpm/blob/2.17-release/plugins/pulp_rpm/plugins/distributors/yum/configuration.py#L376